go/libraries/doltcore/sqle/lookup/rangeslice.go

```diff
@@ -0,0 +1,67 @@
+// Copyright 2020 Dolthub, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package lookup
+
+import "sort"
+
+// RangeSlice is a sortable slice of ranges.
+type RangeSlice []Range
+
+func (r RangeSlice) Len() int      { return len(r) }
+func (r RangeSlice) Swap(i, j int) { r[i], r[j] = r[j], r[i] }
+func (r RangeSlice) Less(i, j int) bool {
+	lc, err := r[i].LowerBound.Compare(r[j].LowerBound)
+	if err != nil {
+		panic(err) //TODO: write own sort that accepts errors
```

Comment:
Please do before shipping. Just make RangeSlice a struct type and store the error in there